common/src/main/java/whocraft/tardis_refined/common/block/console/GlobalConsoleBlock.java

```diff
@@ -95,4 +105,65 @@ public void destroy(LevelAccessor levelAccessor, BlockPos blockPos, BlockState b
 
         super.destroy(levelAccessor, blockPos, blockState);
     }
+
+    @Override
+    public void animateTick(BlockState blockState, Level level, BlockPos blockPos, RandomSource randomSource) {
+        super.animateTick(blockState, level, blockPos, randomSource);
+
+        TardisClientData clientData = TardisClientData.getInstance(level.dimension());
+        if (clientData != null) {
+            if (clientData.isFlying() && level.random.nextInt(4) == 0) {
+                var d = (double)blockPos.getX() + randomSource.nextFloat() * 1.25;
+                var e = (double)blockPos.getY() + randomSource.nextDouble() * 1D + 1D;
+                var f = (double)blockPos.getZ()  + randomSource.nextFloat()* 1.25;
+                level.addParticle(ParticleTypes.CLOUD, d, e, f, 0.0D, 0.1D, 0.0D);
+            }
+
+            if (clientData.isOnCooldown() || clientData.isCrashing()) {
+
+                var d = (double)blockPos.getX() + level.getRandom().nextFloat() * 1.25;
+                var e = (double)blockPos.getY() + level.getRandom().nextDouble() * 1D + 1D;
+                var f = (double)blockPos.getZ()  + level.getRandom().nextFloat()* 1.25;
+                level.addParticle(ParticleTypes.CAMPFIRE_COSY_SMOKE, d, e, f, 0.0D, 0.1D, 0.0D);
+
+                for (int i = 0; i < 5; i++) {
+                     d = (double)blockPos.getX() + level.getRandom().nextFloat() * 1.25;
+                     e = (double)blockPos.getY() + level.getRandom().nextDouble() * 1D + 1D;
+                     f = (double)blockPos.getZ()  + level.getRandom().nextFloat()* 1.25;
+                    level.addParticle(ParticleTypes.LARGE_SMOKE, d, e, f, 0.0D, 0.1D, 0.0D);
+                }
+
+                if (level.random.nextInt(10) == 0) {
+                     d = (double)blockPos.getX() + level.getRandom().nextFloat() * 1.25;
+                     e = (double)blockPos.getY() + level.getRandom().nextDouble() * 1D + 1D;
+                     f = (double)blockPos.getZ()  + level.getRandom().nextFloat()* 1.25;
+                    level.addParticle(ParticleTypes.LAVA, d, e, f, -0.5 + level.random.nextFloat(), 0.05D, -0.5 + level.random.nextFloat());
+                    level.addParticle(ParticleTypes.LAVA, d, e, f, -0.5 + level.random.nextFloat(), 0.05D, -0.5 + level.random.nextFloat());
+                    level.addParticle(ParticleTypes.LAVA, d, e, f, -0.5 + level.random.nextFloat(), 0.05D, -0.5 + level.random.nextFloat());
+                    level.playLocalSound(blockPos.getX(),blockPos.getY(),blockPos.getZ(), SoundEvents.LAVA_EXTINGUISH, SoundSource.BLOCKS, 1, level.getRandom().nextFloat() + 1f, false);
+                }
+            }
+        }
+    }
+
+    @Override
+    public InteractionResult use(BlockState blockState, Level level, BlockPos blockPos, Player player, InteractionHand interactionHand, BlockHitResult blockHitResult) {
+
+        AtomicBoolean shouldConsume = new AtomicBoolean(false);
```

Comment:
We should avoid usage of AtomicBooleans. Instead use isPresent to check if the capability is present and have it as a field instead with the use of get()